packages/vite/rollupLicensePlugin.ts

```diff
@@ -15,79 +16,91 @@ export default function licensePlugin(
       const coreLicense = fs.readFileSync(
         new URL('../../LICENSE', import.meta.url),
       )
-      function sortLicenses(licenses: Set<string>) {
-        let withParenthesis: string[] = []
-        let noParenthesis: string[] = []
-        licenses.forEach((license) => {
-          if (/^\(/.test(license)) {
-            withParenthesis.push(license)
-          } else {
-            noParenthesis.push(license)
+
+      const deps = sortDependencies(dependencies)
+      const licenses = sortLicenses(
+        new Set(
+          dependencies.map((dep) => dep.license).filter(Boolean) as string[],
+        ),
+      )
+
+      let dependencyLicenseTexts = ''
+      for (let i = 0; i < deps.length; i++) {
+        // Find dependencies with the same license text so it can be shared
+        const licenseText = deps[i].licenseText
+        const sameDeps = [deps[i]]
+        if (licenseText) {
+          for (let j = i + 1; j < deps.length; j++) {
+            if (licenseText === deps[j].licenseText) {
+              sameDeps.push(...deps.splice(j, 1))
+              j--
+            }
+          }
+        }
+
+        let text = `## ${sameDeps.map((d) => d.name).join(', ')}\n`
+        const depInfos = sameDeps.map((d) => getDependencyInformation(d))
+
+        // If all same dependencies have the same license and contributor names, show them only once
+        if (
+          depInfos.length > 1 &&
+          depInfos.every(
+            (info) =>
+              info.license === depInfos[0].license &&
+              info.names === depInfos[0].names,
+          )
+        ) {
+          const { license, names } = depInfos[0]
+          const repositoryText = depInfos
+            .map((info) => info.repository)
+            .filter(Boolean)
+            .join(', ')
+
+          if (license) text += `License: ${license}\n`
+          if (names) text += `By: ${names}\n`
+          if (repositoryText) text += `Repository: ${repositoryText}\n`
```

Comment:
Should this be `Repositories: ` or are there tools that will look for "Repository:"?

Reply:
I think that works too. I didn't pay attention to the names when refactoring this part
```suggestion
          if (repositoryText) text += `Repositories: ${repositoryText}\n`
```